### PR TITLE
Add tests for the Header class.

### PR DESCRIPTION
**Pull request by MrGlobby**
These are intended to exercise methods that are currently not covered by other tests according to the coverage report.

### CODE REVIEW

**Review by cary-ilm — APPROVED** [review]
LGTM, except for the test failure.

The CLA should be on the way shortly.

File: src/test/OpenEXRTest/testHeader.cpp [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+        assert(false);
+    }
+};
+
+void testEraseKnownAttribute(const string& name)
+{
+    Header header;
+    header.erase(name);
+    assert(header.find(name) != header.end());
```
Comment (cary-ilm) [inline src/test/OpenEXRTest/testHeader.cpp:64]: Shouldn't that be `header.find(name) == header.end()` since it's been erased?

Comment (MrGlobby) [reply src/test/OpenEXRTest/testHeader.cpp:64]: Oh yes, I'm sorry, I made it != it to ensure that failing tests were being reported for me. For a while at the beginning the assert statements weren't being evaluated and I didn't realise. That was fixed when I realised what the `#ifdef NDEBUG` section at the top of the file was doing, but I left this to make sure things kept reporting correctly and then forgot to correct it before committing.

Comment (cary-ilm) [reply src/test/OpenEXRTest/testHeader.cpp:64]: Thanks, just push a new commit when you're ready.

The CLA should be in place now, too.  Try clicking through the CLA link, with Lucasfilm as the company.

**Comment by cary-ilm** [comment]
Thanks, I'm not sure what's up with the CLA., I was informed it was in place.

File: src/test/OpenEXRTest/testHeader.cpp (hunk 1 above)
Comment (MrGlobby) [reply src/test/OpenEXRTest/testHeader.cpp:64]: Thanks, that's all sorted now I think!

**Review by cary-ilm — APPROVED** [review]
LGTM, thanks!

**Pull request merged**